lib/components/Forms/Fields/Input/index.tsx

```diff
@@ -1,11 +1,13 @@
 export * from "@/ui/input"
 import { Input as ShadcnInput } from "@/ui/input"
-import { ComponentProps } from "react"
+import { ComponentProps, HTMLInputTypeAttribute } from "react"
 
-type InputProps = Pick<
+export type InputProps = Pick<
   ComponentProps<typeof ShadcnInput>,
-  "ref" | "type" | "disabled" | "size" | "onChange" | "value" | "placeholder"
->
+  "ref" | "disabled" | "size" | "onChange" | "value" | "placeholder"
+> & {
+  type?: Exclude<HTMLInputTypeAttribute, "number">
```

Comment:
Why should we ban a `number` type? I understand they intersect with the `NumberInput`, but they are not the same.

`NumberInput` does not cover the case when you have a counter with steps (classic `type="number"` use case)

I think we should keep the number type available. 

@josepjaume wdyt?

Reply:
> Why should we ban a number type? I understand they intersect with the NumberInput, but they are not the same.

If we allow both, we'll have different UX across the app as pointed out [here](https://github.com/factorialco/factorial/pull/47179), so the idea is to have a single way to render a numeric input to avoid those issues.

> NumberInput does not cover the case when you have a counter with steps (classic type="number" use case)

The gamma version has the steps and I just added it for this one as well. I opened the PR some weeks ago without it because at that time I didn't manage to add the icons for some reason that I don't remember 😅 

https://github.com/user-attachments/assets/d2998f77-16a1-4ec4-9b39-cca421459196

Reply:
Interesting! I'm also struggling to decide to which extent we want to mask the underlying `input` props in our components or not. I'm leaning a bit more towards @viniciusmeneses's approach. passing the `type` to `input` feels like it belongs to an underlying layer.

Should we explicitly provide `PasswordInput`, `TextInput` and things like these instead (following the `NumberInput` example)? They convey more meaning and one could see their UX evolving in slightly different ways (passwords having a built-in complexity validation, numbers having arrows...).

Reply:
I'd rather rendered `NumberInput` when an engineer types `<Input type="number">`, but both approaches work as long as they are consistently applied.

It would be nice to somehow notify the user about the alternative

Reply:
The problem with this @nlopin is that I can't see how TypeScript would apply the right types. The signature of the props of `<NumberInput />` will potentially be different from `Input` in the mid term (that's why we're creating a separate component) so that would get in the middle.

Reply:
> ...passing the type to input feels like it belongs to an underlying layer.

> Should we explicitly provide PasswordInput, TextInput and things like these instead (following the NumberInput example)?

💯 agree with this approach. I prefer separate components that provide a more constrained set of handles that impact what we are trying to do specifically in that context. It lets our interfaces read cleaner (values, handlers, etc.) and the implementation can be more cleanly maintained than in a component that handles each case or internally delegates to another component (which gets us back to this).

> The signature of the props of <NumberInput /> will potentially be different from Input in the mid term

*Technically* we could support this using Union Types, but not my favorite. In the following excerpt TS will correctly infer that `maxDecimals` is not supported when type is `password` and the type of the incoming value for `onChange`:

![Screenshot 2024-09-12 at 10 55 45](https://github.com/user-attachments/assets/94dc6aec-4188-4d25-8493-e6999c8cf138)

Reply:
> Technically we could support this using Union Types


Yeah, union types will always be a PITA (autocompletes are awkward) and we'd end up having a pretty complicated bunch of conditionals to distinguish components. It feels to me that going the other way around is going to be more effective: separate entrypoints, and sharing logic as we see fit in their implementations (as opposed to sharing logic by default and then having to artifficially separate the code paths).

Reply:
💯